### DISCUSSION TITLE
Screen-sharing with Regolith 3.0 on Wayland not working.

### DISCUSSION THREAD

**Discussion by m-wachter** [regolith-linux/regolith-desktop#914] [Q&A] [answered]
Hello,

I did a fresh install of Ubuntu 13.04 with the "normal" Ubuntu desktop and installed Regolith 3.0 as described here https://regolith-desktop.com/docs/using-regolith/install/ with `sudo apt install regolith-desktop regolith-session-sway regolith-look-nord`

I tired to do screen-sharing with jitsi on firefox, but no dialog to choose the screen or the window to share is shown.

The journal shows: 
```
Sep 17 16:35:30 carver xdg-desktop-por[11736]: A backend call failed: Keine derartige Schnittstelle »org.freedesktop.impl.portal.ScreenCast« des Objekts im Pfad /org/freedesktop/portal/desktop
Sep 17 16:35:30 carver xdg-desktop-por[11736]: Failed to close session implementation: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: Das Objekt existiert nicht am Pfad »/org/freedesktop/portal/desktop/session/1_97/webrtc_session924246257« 
```

Translated manually:
```
Sep 17 16:35:30 carver xdg-desktop-por[11736]: A backend call failed: No interface »org.freedesktop.impl.portal.ScreenCast« of the object at path /org/freedesktop/portal/desktop
Sep 17 16:35:30 carver xdg-desktop-por[11736]: Failed to close session implementation: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: the object does not exist at path »/org/freedesktop/portal/desktop/session/1_97/webrtc_session924246257« 
```

What am I doing wrong ?

Greetings

Michael.

**Comment by Sturgelose** [comment]
I'm having a similar issue. I'm using latest release available in 23.04 and neither chrome or firefox are showing me gnome-portal to choose which screen to share.

I definitely have it installed but nothing is really happening....

**Comment by SoumyaRanjanPatnaik** [comment]
Do you have pipewire as the default sound server? Pipewire is a requirement for screensharing to work on wayland.

**Reply by Sturgelose** [reply]
Yes, I do have it installed and running. If I try to run pipewire manually, it tells me it is already running (there is a lock in place). 

In fact, the weird thing is that in Ubuntu Wayland I can share screen without any issue (aside the fact that it requires me to choose every time which window to share, but that is a known issue in Gnome-portal) in both chrome and firefox

However in regolith-sway, this is not happening (for both browsers). I can videocall (video and sound works), but screen sharing is failing miserably. No error at all or at least I haven't been able to find the log telling me exactly what happened.

So, it feels there some kind of difference between regolith-sway and ubuntu wayland (23.04) whenever sharing screen....

Also, if it helps: flameshot works properly in ubuntu wayland, but in regolith-sway it also fails. Not sure if it is related or unrelated though....

**Reply by SoumyaRanjanPatnaik** [reply, marked as answer]
try this: 

In `~/.config/environment.d/sway_env.conf`, add `XDG_CURRENT_DESKTOP=sway`, followed by a logout. Check if screensharing works the next time you log in. Use [this](https://mozilla.github.io/webrtc-landing/gum_test.html) to test.

**Note**: Screensharing uses `slurp` by default, which requires you to click the display you want to share.

**Reply by Sturgelose** [reply]
@SoumyaRanjanPatnaik this definitely did the trick and works perfect! :1st_place_medal: 

I had a bit of a hunch this was the issue because in flameshot they mentioned this specific ENV variable but I had no idea where it should be set.

If there is anything to comment: `slurp` selection is perfect, even easier than the default gnome-portal in ubuntu! However, I still need to select the window/screen once, choose it, and then click on it again. It's kinda if like I have to select it twice (but I guess it might be an issue with Chrome/Chromium) and with gnome-portal I have a similar behaviour.

Do you think it would make sense to somehow set the `XDG_CURRENT_DESKTOP=sway` as default to have better compatibility? I understand `regolith-sway` is more specific, but at least, right now, software is using it to detect its wayland/sway compatibility :sweat_smile:

**Reply by SoumyaRanjanPatnaik** [reply]
I think it definitely makes sense to keep this as the default.

**Reply by cfsmp3** [reply]
Be aware that flameshot doesn't work well - specifically, you can save your selection to file (and that works perfectly), but copying to the clipboard does not.

**Reply by Sturgelose** [reply]
I'm aware :) I'll be trying to configure today https://github.com/jtheoof/swappy, that together with slurp and grim you can get something similar to fireshot.

Otherwise there is no working out-of-the-box configuration for screenshots in regolith-sway!

**Comment by jc00ke** [comment]
Is there no way to share just a window?

**Reply by SoumyaRanjanPatnaik** [reply]
Not yet. sway doesn't support this yet.

**Reply by jc00ke** [reply]
Hi there, sorry to necro-post, but is this supported now? Thanks!

**Comment by mzeo** [comment]
This does not work for me. Firefox says I need to grant permission.
```
▶ cat ~/.config/environment.d/sway_env.conf
XDG_CURRENT_DESKTOP=sway

```
